src/hooks/useKanban.js

```diff
@@ -318,6 +319,71 @@ visitorId = 'visitor_' + Date.now() + '_' + Math.random().toString(36).substr(2,
     }
   };
 
+  // Adiciona uma label a uma task
+  const addLabel = async (columnId, taskId, label) => {
+    if (!user) return;
+    
+    try {
+      console.log('Adicionando label:', { columnId, taskId, label });
+      const column = columns[columnId];
+      if (!column) {
+        console.error('Coluna não encontrada:', columnId);
+        return;
+      }
+
+      const taskIndex = column.tasks.findIndex(t => t.id === taskId);
+      if (taskIndex === -1) {
+        console.error('Task não encontrada:', taskId);
+        return;
+      }
+
+      const task = column.tasks[taskIndex];
+      console.log('Task atual:', task);
+      
+      const updatedTask = {
+        ...task,
+        labels: [...(task.labels || []), {
+          id: crypto.randomUUID(),
+          text: label.text,
+          color: label.color
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
if (!label?.text || !label?.color) {
  throw new Error('Invalid label format: text and color are required');
}
// Then proceed with the label addition
```
The addLabel function doesn't validate the label object structure. Add input validation to ensure label.text and label.color exist and are of correct types.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;